__test__/server/api/createOptOut.test.js

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable no-unused-expressions, consistent-return */
+import { r } from "../../../src/server/models/";
+
+import {
+  setupTest,
+  cleanupTest,
+  runComponentGql,
+  createUser,
+  createInvite,
+  createOrganization,
+  createCampaign,
+  createContacts,
+  createTexter,
+  assignTexter,
+  startCampaign,
+  getCampaignContact,
+  getOptOut
+} from "../../test_helpers";
+
+import { createOptOutMutation } from "../../../src/components/IncomingMessageList/ConversationPreviewModal";
+
+describe.only("createOptOut", () => {
+  let testAdminUser;
+  let testInvite;
+  let testOrganization;
+  let testCampaign;
+  let testTexterUser;
+  let testTexterUser2;
+  let testContacts;
+  let assignmentId;
+  let organizationId;
+  let optOutContact;
+  let optOut;
+  let variables;
+
+  beforeEach(async () => {
+    // Set up an entire working campaign
+    await setupTest();
+    testAdminUser = await createUser();
+    testInvite = await createInvite();
+    testOrganization = await createOrganization(testAdminUser, testInvite);
+    testCampaign = await createCampaign(testAdminUser, testOrganization);
+    testContacts = await createContacts(testCampaign, 100);
+    testTexterUser = await createTexter(testOrganization);
+    testTexterUser2 = await createTexter(testOrganization);
+    await startCampaign(testAdminUser, testCampaign);
+
+    await assignTexter(testAdminUser, testTexterUser, testCampaign);
+    const dbCampaignContact = await getCampaignContact(testContacts[0].id);
+    assignmentId = dbCampaignContact.assignment_id;
+    organizationId = testOrganization.data.createOrganization.id;
+
+    optOutContact = testContacts[20];
+    optOut = {
+      cell: optOutContact.cell,
+      assignmentId,
+      reason: "they were snotty"
+    };
+
+    variables = {
+      optOut,
+      campaignContactId: optOutContact.id,
+      organizationId: organizationId.toString()
+    };
+  }, global.DATABASE_SETUP_TEARDOWN_TIMEOUT);
+
+  afterEach(async () => {
+    await cleanupTest();
+    if (r.redis) r.redis.flushdb();
+  }, global.DATABASE_SETUP_TEARDOWN_TIMEOUT);
+
+  it("creates an opt out when the contact is assigned to the current user", async () => {
+    const optOutResult = await runComponentGql(
```

Comment:
This test got me thinking -- I'd actually like all of our optOut calls to just have an `id` field and not make anything else available.  So, I'd rather we test existence by seeing the id attribute come back (maybe always the same thing, even).

I.e. this line should always be an acceptable result:
https://github.com/MoveOnOrg/Spoke/blob/main/src/server/api/campaign-contact.js#L192
(and maybe we should make it a single char 'x' or 'y' or something)

Reply:
@schuyler1d let's punt this here.  See #1314.